PZNS_Framework/media/lua/client/03_mod_core/PZNS_NPCSurvivor.lua

```diff
@@ -1,61 +1,204 @@
-        npcIsoPlayerObject = npcIsoPlayerObject -- Cows: objects cannot be saved to moddata...
+        isAlive = true,
+        isSpawned = true,
+        forename = "",
+        surname = "",
+        isFemale = false,
+        squareX = nil,
+        squareY = nil,
+        squareZ = nil,
+        npcIsoPlayerObject = npcIsoPlayerObject
     };
 
+    setmetatable(npcSurvivor, self);
+    self.__index = self;
+
     return npcSurvivor;
 end
+
+---Assigns groupID to NPC
+---@param groupID groupID
+function NPC:setGroupID(groupID)
+    self.groupID = groupID
+end
+
+---Unassign groupID from NPC
+function NPC:unsetGroupID()
+    self.groupID = nil
+end
+
+---comments
+---@param mpPlayerID integer
+function NPC:setPlayerNum(mpPlayerID)
+    local requestedIsoPlayer = getSpecificPlayer(mpPlayerID)
+    assert(requestedIsoPlayer, "Can't find specific player, num: " .. tostring(mpPlayerID))
+    local msg = string.format("IsoPlayer for mpPlayerID: %s does not match npcIsoPlayerObject. survivorID: %s",
+        mpPlayerID, self.survivorID)
+    assert(requestedIsoPlayer == self.npcIsoPlayerObject, msg)
+    self.mpPlayerID = mpPlayerID
+end
+
+--region relations
+
+---Change relation with `otherSurvivorID` by `diff`
+---@param otherSurvivorID survivorID survivor to change relation to
+---@param diff integer change amount
+---@return boolean isNewRelation whether this relation is new (first meet)
+function NPC:changeRelation(otherSurvivorID, diff)
+    --TODO: periodic(?) cleanings for nil NPC (dead etc)
+    local relation = self.relationsMap[otherSurvivorID]
+    local isNewRelation = false
+    local newRelation
+    if not relation then
+        local anonOffset = self.anonymousMap[otherSurvivorID] or 0
+        relation = self.initialRelation + anonOffset
+        isNewRelation = true
+        self.anonymousMap[otherSurvivorID] = nil
+    end
+    newRelation = relation + diff
+    newRelation = math.min(self._maxRelation, math.max(self._minRelation, newRelation))
+    self.relationsMap[otherSurvivorID] = newRelation
+    return isNewRelation
+end
+
+---Change anonymous relation with `otherSurvivorID` by `diff`
+---@param otherSurvivorID any
+---@param diff integer
+---@return boolean isNewRelation whether this anon relation is new (first seen)
+function NPC:changeAnonRelation(otherSurvivorID, diff)
+    local relation = self.anonymousMap[otherSurvivorID]
+    local newRelation
+    local isNewRelation = false
+    if not relation then
+        relation = self.initialRelation
+        isNewRelation = true
+    end
+    newRelation = relation + diff
+    newRelation = math.min(self._maxRelation, math.max(self._minRelation, newRelation))
+    self.anonymousMap[otherSurvivorID] = newRelation
+    return isNewRelation
+end
+
+---Get relation to `otherSurvivorID`
+---@param otherSurvivorID survivorID
+---@return integer?
+function NPC:getRelationTo(otherSurvivorID)
```

Comment:
Well, this is interesting... I didn't consider going deeper into NPC-to-NPC relations as it seemed to be too much work.

Reply:
This was just an idea, not really thought through enough.